helpers/base.py

```diff
@@ -87,3 +87,30 @@ def in_network(ip_address, cidrs):
         if ip_address in network:
             return True
     return False
+
+def fetch_values_by_datatype(rec, datatype):
+    """Fetch values of normalized_type.
+
+    Args:
+        rec (dict): parsed payload of any log
+        datatype (str): normalized type user interested
+
+    Returns:
+        (list) The values of normalized types
+    """
+    results = []
+    if not datatype in rec['normalized_types'].keys():
+        return results
+
+    for key in rec['normalized_types'][datatype]:
+        # Normalized type may be in nested subkeys, we only support one level of
```

Comment:
To confirm, this will work for cases where `envelope_keys` is used, correct?

Reply:
Based on my live test (which has `envelope_keys` defined), it works for cases where `envelope_keys` is used.